prow/github/client.go

```diff
@@ -89,19 +90,36 @@ func (c *Client) log(methodName string, args ...interface{}) {
 	c.Logger.Printf("%s(%s)", methodName, strings.Join(as, ", "))
 }
 
-// Retry on transport failures. Does not retry on 500s.
+var timeSleep = time.Sleep
+
+// Retry on transport failures. Retries on 500s and retries after sleep on
+// ratelimit exceeded.
 func (c *Client) request(method, path string, body interface{}) (*http.Response, error) {
 	var resp *http.Response
 	var err error
 	backoff := retryDelay
 	for retries := 0; retries < maxRetries; retries++ {
 		resp, err = c.doRequest(method, path, body)
 		if err == nil {
-			break
+			// If we are out of API tokens, sleep first. The X-RateLimit-Reset
```

Comment:
We will pile up if there are more requests coming in then we have quota for. A bug causing us to spam GitHub will cause them to pile up without limit.

Velodrome currently monitors our token usage, and it's probably a good idea for it to alert if we're ever close to running out. For search quota, it should probably alert if we're running out for 10+ minutes in a row or something.